src/Microsoft.AspNetCore.Razor/TagHelpers/DefaultTagHelperContent.cs

```diff
@@ -282,29 +277,19 @@ private bool IsEmptyOrWhiteSpaceCore(object entry, EmptyOrWhiteSpaceWriter write
         {
             if (entry == null)
             {
-                return false;
+                return true;
             }
 
-            var stringValue = entry as string;
-            if (stringValue != null)
+            if (entry is string stringValue)
             {
                 // Do not encode the string because encoded value remains whitespace from user's POV.
-                if (!string.IsNullOrWhiteSpace(stringValue))
```

Comment:
@NTaylorMullen I've done the mentioned additional changes to this method. Looks better to me. Agreed?